packages/contracts/contracts/L1/rollup/CanonicalTransactionChain.sol

```diff
@@ -390,36 +380,17 @@ contract CanonicalTransactionChain is ICanonicalTransactionChain, Lib_AddressRes
 
             // Now process any subsequent queue transactions.
             nextQueueIndex += uint40(curContext.numSubsequentQueueTransactions);
-
         }
+
         require(
-            nextQueueIndex <= queueLength,
+            nextQueueIndex <= queueElements.length,
             "Attempted to append more elements than are available in the queue."
         );
 
         // Generate the required metadata that we need to append this batch
         uint40 numQueuedTransactions = totalElementsToAppend - numSequencerTransactions;
-        uint40 blockTimestamp;
-        uint40 blockNumber;
-        if (curContext.numSubsequentQueueTransactions == 0) {
-            // The last element is a sequencer tx, therefore pull timestamp and block number from
-            // the last context.
-            blockTimestamp = uint40(curContext.timestamp);
-            blockNumber = uint40(curContext.blockNumber);
-        } else {
-            // The last element is a queue tx, therefore pull timestamp and block number from the
-            // queue element.
-            // curContext.numSubsequentQueueTransactions > 0 which means that we've processed at
-            // least one queue element. We increment nextQueueIndex after processing each queue
-            // element, so the index of the last element we processed is nextQueueIndex - 1.
-            Lib_OVMCodec.QueueElement memory lastElement = _getQueueElement(
-                nextQueueIndex - 1,
-                queueRef
-            );
-
-            blockTimestamp = lastElement.timestamp;
-            blockNumber = lastElement.blockNumber;
-        }
```

Comment:
Since we no longer have the `timestamp` and `blockNumber` available on the last queue element in the batch, instead of having an if/else to determine if we should use the values from the last queue element, we always use the current values from the EVM. I think this logic was actually just 'vestigial' monotonicity code, that could have been removed anyways, but would appreciate you taking a look there.